# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/build/components.conf
# Commit: 201f0d5fabd6
# Full Hash: 201f0d5fabd61c8eebe7e27e8597021d3618ddea
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2021-03-29 15:25:22
# Regressor Bug: 1695050
# File Overlap Count: 1
# Description:
#   Bug 1695050 - Part 2: Modify CookieJarSettings::Create() to be able to creat cookieJarSettings according to the browsing mode. r=dimi,preferences-reviewers
#   
#   This patch modifies the current CookieJarSettings::Create() function.
#   It removes the current function and adds two variants. One takes the
#   nsIPrincipal as input and another takes the enum value. The new
# ==============================================================================

diff -r 59cda3c92f48 -r 201f0d5fabd6 netwerk/build/components.conf
--- a/netwerk/build/components.conf	Mon Mar 29 11:01:50 2021 +0000
+++ b/netwerk/build/components.conf	Mon Mar 29 11:01:51 2021 +0000
@@ -614,7 +614,7 @@
         'cid': '{4ce234f1-52e8-47a9-8c8d-b02f815733c7}',
         'contract_ids': ['@mozilla.org/cookieJarSettings;1'],
         'type': 'nsICookieJarSettings',
-        'constructor': 'mozilla::net::CookieJarSettings::Create',
+        'constructor': 'mozilla::net::CookieJarSettings::CreateForXPCOM',
         'headers': ['mozilla/net/CookieJarSettings.h'],
     },
 ]